res/enkf/local_config.py

```diff
@@ -13,120 +13,124 @@
-
+        if mini_step_key in [ministep.name() for ministep in self.ministeps]:
+            raise KeyError(f"Ministep: {mini_step_key} already in local_config")
+        return self.add_ministep(mini_step_key)
+
+    @deprecated(
+        "Add observations to ministep directly",
+        removed_in="2.37",
+        current_version=__version__,
+    )
     def createObsdata(self, obsdata_key):
         """@rtype: Obsdata"""
-        assert isinstance(obsdata_key, str)
-        if self._has_obsdata(obsdata_key):
-            raise ValueError(f"Tried to add existing observation key: {obsdata_key}")
-
-        return _lib.local.local_config.create_obsdata(self, obsdata_key)
+        return LocalObsdata(obsdata_key)
 
-    def copyObsdata(self, src_key, target_key):
-        """@rtype: Obsdata"""
-        assert isinstance(src_key, str)
-        assert isinstance(target_key, str)
-        if not self._has_obsdata(src_key):
-            raise KeyError(f"The observation set {src_key} does not exist")
-
-        obsdata = _lib.local.local_config.get_obsdata_copy(self, src_key, target_key)
-        return obsdata
+    @deprecated(
+        "Add observations to ministep directly",
+        removed_in="2.37",
+        current_version=__version__,
+    )
+    def copyObsdata(self, _, new_key):
+        return LocalObsdata(new_key)
 
+    @deprecated(
+        "Loop over configuration directly",
+        removed_in="2.37",
+        current_version=__version__,
+    )
     def getUpdatestep(self):
         """@rtype: UpdateStep"""
-        return self._get_updatestep()
+        return self
 
-    def getMinistep(self, mini_step_key):
+    @deprecated(
+        "Add observations to ministep directly",
+        removed_in="2.37",
+        current_version=__version__,
+    )
+    def getMinistep(self, key):
         """@rtype: Ministep"""
-        assert isinstance(mini_step_key, str)
-        return self._get_ministep(mini_step_key)
-
+        for ministep in self.ministeps:
+            if ministep.name() == key:
+                return ministep
+        raise KeyError(f"No ministep named: {key}")
+
+    @deprecated(
+        "Add observations to ministep directly",
+        removed_in="2.37",
+        current_version=__version__,
+    )
     def getObsdata(self, obsdata_key):
         """@rtype: Obsdata"""
-        assert isinstance(obsdata_key, str)
-        return _lib.local.local_config.get_obsdata_ref(self, obsdata_key)
+        return obsdata_key
+
+    @deprecated("Use add_ministep", removed_in="2.37", current_version=__version__)
+    def attachMinistep(self, mini_step):
+        if mini_step.name not in [ministep.name for ministep in self.ministeps]:
+            self.ministeps.append(mini_step)
 
-    def attachMinistep(self, update_step, mini_step):
-        assert isinstance(mini_step, LocalMinistep)
-        assert isinstance(update_step, LocalUpdateStep)
-        self._attach_ministep(update_step, mini_step)
+    def add_ministep(self, name):
```

Comment:
Maybe ministeps should be created complete, i.e. take in a set of parameters and observations in the constructor. No partially valid objects